pkg/crd/crd.go

```diff
@@ -0,0 +1,143 @@
+// Package crd handles the dynamic creation and modification of CustomResourceDefinitions.
+package crd
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	clientv1 "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var (
+	waitDuration = 1 * time.Minute
```

Comment:
Feel free to ignore this nit - but I would make these constants here and pass them to your waitForCRD function. They seem to be variables just so they can be changed in tests, which is OK, but not ideal.

Reply:
Since `time.Minute` is technically a struct the compiler won't allow me to make it constant.

Reply:
It will, since values of type `time.Duration` are really just numbers.

Reply:
Oh nice, I didn't think it would.  I am going to leave it since it is not exported, and it keeps the test from taking a minute to run when testing the public `BatchCreate(..)` function.

Reply:
I was thinking you could pass it as a parameter, rather than use the package variable, but it's up to you.

Reply:
Oh, you mean to the `BatchCreateCRDs()` sure I am fine with that.

Reply:
Or rather `waitCRD` could take it. And `BatchCreateCRDs` would pass it...